### PR TITLE
Remove deprecated SYCL usage for newer compilers

### PR DESCRIPTION
**Pull request by rhornung67**
# Summary

- This PR squashes a bunch of compiler warnings due to deprecated SYCL usage in camp and RAJA. 
- Note: this PR depends on changes in a camp branch. That needs to be merged and pulled in here before this can be merged.

### CODE REVIEW

**Comment by rhornung67** [comment]
This is good to merge. Please review when you have some time. Thank you.

File: include/RAJA/policy/sycl/policy.hpp [hunk 1]
```diff
@@ -39,7 +39,7 @@ struct uint3 {
   unsigned long x, y, z;
 };
 
-using sycl_dim_t = cl::sycl::range<1>;
+using sycl_dim_t = sycl::range<1>;
```
Comment (rchen20) [inline include/RAJA/policy/sycl/policy.hpp:42]: Maybe keep the global scope `::sycl` for consistency?

```suggestion
using sycl_dim_t = ::sycl::range<1>;
```

Comment (rhornung67) [reply include/RAJA/policy/sycl/policy.hpp]: Good catch. Fixed.

Note that I didn't exhaustively search the code for these. I only changed the lines that generated compiler warnings.

File: include/RAJA/policy/sycl/reduce.hpp [hunk 2]
```diff
@@ -107,11 +107,11 @@ struct Reduce_Data
...
       : value(initValue)
   {
-    cl::sycl::queue* q = ::camp::resources::Sycl::get_default().get_queue();
+    ::sycl::queue* q = ::camp::resources::Sycl::get_default().get_queue();
 
 
-    device = reinterpret_cast<T *>(cl::sycl::malloc_device(sycl::MaxNumTeams * sizeof(T), *(q)));
-    host = reinterpret_cast<T *>(cl::sycl::malloc_host(sycl::MaxNumTeams * sizeof(T), *(q)));
+    device = reinterpret_cast<T *>(::sycl::malloc_device(sycl::MaxNumTeams * sizeof(T), *(q)));
+    host = reinterpret_cast<T *>(::sycl::malloc_host(sycl::MaxNumTeams * sizeof(T), *(q)));
```
Comment (rchen20) [inline include/RAJA/policy/sycl/reduce.hpp:114]: The references to `MaxNumTeams` look slightly confusing; should they be under the `RAJA::sycl::` namespace?

```suggestion
    device = reinterpret_cast<T *>(::sycl::malloc_device(RAJA::sycl::MaxNumTeams * sizeof(T), *(q)));
    host = reinterpret_cast<T *>(::sycl::malloc_host(RAJA::sycl::MaxNumTeams * sizeof(T), *(q)));
```

Comment (rhornung67) [reply include/RAJA/policy/sycl/reduce.hpp:114]: Note that I did not write this code and was fixing compiler warnings. It looks like this file was started as a copy of the similarly named openmp target file.  

This is probably a better was to define the variable: https://github.com/LLNL/RAJA/blob/develop/include/RAJA/policy/openmp_target/reduce.hpp#L70

However, that file also contains similar namespace scoping issues.

I'll make an issue and we can fix those files and others for consistency as needed in a separate PR.

**Review by artv3 — APPROVED** [review]

**Review by rchen20 — APPROVED** [review]

**Comment by artv3** [comment]
I'm having the same build failure (OMP Target) on a different RAJA PR....

**Comment by rhornung67** [comment]
> I'm having the same build failure (OMP Target) on a different RAJA PR....

Try restarting the failed job

**Pull request merged**